docs/guidelines/guidelines.md

```diff
@@ -0,0 +1,158 @@
+
+````md
+- Example of a Istio resource:
+```yaml { ignore=true }
+apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  namespace: istio-system
+  updateStrategy:
+    type: RevisionBased
+    inactiveRevisionDeletionGracePeriodSeconds: 30
+  version: v1.24.2
+```
+````
+
+*Note:* take into account that all the code blocks that are going to be run by the `runme` tool should be tagged with the language `bash` at the beginning of the block and they need to contain only commands that are going to be run in the terminal and all the steps that are not needed to be executed because they are not commands or they are not needed for the goal of the example should be ignored with the `ignore=true` tag. For example
+
+````md
+- You shouuld set version for Istio in the `Istio` resource and `IstioRevisionTag` resource should reference the name of the `Istio` resource:
+```yaml { ignore=true }
+apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  namespace: istio-system
+  updateStrategy:
+    type: RevisionBased
+    inactiveRevisionDeletionGracePeriodSeconds: 30
+  version: v1.24.2
+---
+apiVersion: sailoperator.io/v1
+kind: IstioRevisionTag
+metadata:
+  name: default
+spec:
+  targetRef:
+    kind: Istio
+    name: default
+```
+- Create ns, `Istio` and `IstioRevisionTag` resources:
+```bash { name=create-istio tag=example}
+kubectl create ns istio-system
+cat <<EOF | kubectl apply -f-
+apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  namespace: istio-system
+  updateStrategy:
+    type: RevisionBased
+    inactiveRevisionDeletionGracePeriodSeconds: 30
+  version: v1.24.2
+---
+apiVersion: sailoperator.io/v1
+kind: IstioRevisionTag
+metadata:
+  name: default
+spec:
+  targetRef:
+    kind: Istio
+    name: default
+EOF
+```
+````
+
+- Validation steps are always needed, to add a validation step you can comment directly the entire code block with the bash command that you want to run. This validation steps need to use the same tag as the example where they are going to run and should be prefixed with `validation-description`. For example:
+```md
+<!-- ```bash { name=validation-wait-operator tag=example}
+kubectl wait --for=condition=available --timeout=600s deployment/sail-operator -n sail-operator
+``` -->
+```
+
+*Important:* please add always validation steps to avoid flakyness during the execution, this steps will ensure that resource conditions are met and the test will not fail because of a timeout or any other issue. This validation needs to add failures to the test if the condition is not met.
```

Comment:
```suggestion
*Important:* please always add validation steps to avoid flakyness during the execution, these steps will ensure that resource conditions are met and the test will not fail because of a timeout or any other issue. This validation needs to fail the test if the condition is not met, by returning a non-zero return code. Use `kubectl wait` over `sleep` wherever possible.
```